### PR TITLE
Multilanguage simulator for Companion and Firmware simulator 2.1 #2895

### PR DESCRIPTION
**Pull request by fred99**
For test, the translation is limited to 4 languages: SET (translation FR IT DE EN)
companion\src\simulation\simulatorinterface.cpp has been developped by @Sacre100

### CODE REVIEW

File: companion/src/firmwares/opentx/simulator/CMakeLists.txt [hunk 1]
```diff
@@ -1,26 +1,34 @@
+if (TRANSLATIONS STREQUAL "EN") 
+    set (SUFFIX "")
+    MESSAGE( "Generating -en")
+    else ()
```
Comment (projectkk2glider) [inline companion/src/firmwares/opentx/simulator/CMakeLists.txt]: Indentation problem

File: companion/src/firmwares/opentx/simulator/CMakeLists.txt [hunk 2]
```diff
@@ -1,26 +1,34 @@
...
-add_custom_target(commondeps
+add_custom_target(commondeps${SUFFIX}
   make stamp_header tra
   WORKING_DIRECTORY ${RADIO_SRC_DIRECTORY}
 )
 
 set_directory_properties(PROPERTIES ADDITIONAL_MAKE_CLEAN_FILES ${RADIO_SRC_DIRECTORY}/stamp-opentx.h)
 
-add_custom_target(lua_exports 
+add_custom_target(lua_exports${SUFFIX}
```
Comment (projectkk2glider) [inline companion/src/firmwares/opentx/simulator/CMakeLists.txt]: Do we need Lua exports for different languages?

**Comment by fred99** [comment]
Yes, wee need it because of the loop.
If I remove the SUFFIX, I get:
Generating -fr
Generating -it
CMake Error at firmwares/opentx/simulator/CMakeLists.txt:31 (add_custom_target):
  add_custom_target cannot create target "lua_exports" because another target
  with the same name already exists.  The existing target is a custom target
  created in source directory
  "I:/opentx1/companion/src/firmwares/opentx/simulator".  See documentation
  for policy CMP0002 for more details.


Generating -de
CMake Error at firmwares/opentx/simulator/CMakeLists.txt:31 (add_custom_target):
  add_custom_target cannot create target "lua_exports" because another target
  with the same name already exists.  The existing target is a custom target
  created in source directory
  "I:/opentx1/companion/src/firmwares/opentx/simulator".  See documentation
  for policy CMP0002 for more details.


Generating -en
CMake Error at firmwares/opentx/simulator/CMakeLists.txt:31 (add_custom_target):
  add_custom_target cannot create target "lua_exports" because another target
  with the same name already exists.  The existing target is a custom target
  created in source directory
  "I:/opentx1/companion/src/firmwares/opentx/simulator".  See documentation
  for policy CMP0002 for more details.

File: companion/src/firmwares/opentx/simulator/CMakeLists.txt (hunk 1 above)
Comment (fred99) [inline companion/src/firmwares/opentx/simulator/CMakeLists.txt]: I have corrected.

**Comment by fred99** [comment]
The indentation problem is corrected.

File: companion/src/CMakeLists.txt [hunk 3]
```diff
@@ -118,7 +118,15 @@ add_subdirectory(shared)
 add_subdirectory(modeledit)
 add_subdirectory(generaledit)
 add_subdirectory(simulation)
-add_subdirectory(firmwares/opentx/simulator)
+
+SET (translation FR IT DE EN)
```
Comment (bsongis) [inline companion/src/CMakeLists.txt]: No this is not what I would like to see. I would prefer on demand compilation, just as we do for firmwares (you will download your simu)

**Comment by fred99** [comment]
I hope that our Companion modifications are reusable with your "on demand compilation mechanism" otherwise all our "contribution" is just good for trash.
At least, I created a local French workaround, so I'm not just stupid in this story.

**Comment by bsongis** [comment]
I remember I explained what I wanted to do in a post before! It's just impossible to compile all flavors for the firmware and the same applies to the simulators!

**Comment by fred99** [comment]
May be I miss something, but for the simulators, we have "only" 12 flavors = 12 DLL linked to radio type and not the hudge number of firmware flavors linked to the combination of -+ 15 possible options  Isn't it ?

**Comment by kilrah** [comment]
Currently the simulators have a "frozen" set of options that may not match what you like to use on the radio.

The goal is for companion to compile and download the simulator on demand with the same options you've configured for the firmware (including language) so you get to simulate exactly what you'll get on the radio.

**Comment by fred99** [comment]
Thanks for the explenations, the goal goes further than my initial enhancement request: the title should be renamed multilanguage and multi-option simulator ....
I can't help on this new objective.
In the meantime, I will maintain my workaround on the http://frskytaranis.forumactif.org forum. We can't select options, no  problem, we already have all necessary, but a least, we have a simu in french.

**Comment by fred99** [comment]
This branch doesn't answer to the ambitious goal to have a multilanguage and multi-options simulator and can't be a candidat for a merge.

**Pull request closed without merging**